app.js

```diff
@@ -19,6 +22,24 @@ app.use(bodyParser.urlencoded({
   extended: true,
 }));
 
+
+app.use(express.static(path.join(__dirname, "public")));
+app.use(session({
+  secret: 'this is not a secret ;)',
+  resave: false,
+  saveUninitialized: false
+  }));
+app.use(passport.initialize());
```

Comment:
you are missing the line `passport.use(new LocalStrategy(User.authenticate()));`
This wouldn't work unless you also require the User model.